File: src/types/product.ts
export interface Product {
  id: string;
  sku: string;
  name: string;
  slug: string;
  description?: string;
  category_id: string;
  price: number;
  cost_price?: number;
  tax_rate?: number;
  main_image?: string;
  images?: string[];
  available_sizes?: string[];
  size_guide_html?: string;
  is_active: boolean;
  is_featured: boolean;
  barcode?: string;
  view_count: number;
  purchase_count: number;
  average_rating: number;
  created_at: Date;
  updated_at: Date;
}

export interface ProductFilters {
  category_id?: string;
  search?: string;
  min_price?: number;
  max_price?: number;
  is_featured?: boolean;
  page?: number;
  limit?: number;
}

File: apps/pos/Models/Product.cs
using System;
using System.Collections.Generic;

namespace pos.Models;

public class Product
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Guid CategoryId { get; set; }
    public decimal Price { get; set; }
    public decimal CostPrice { get; set; }
    public decimal TaxRate { get; set; } = 0.17m; // 17% default
    public string MainImage { get; set; } = string.Empty;
    public string Images { get; set; } = string.Empty; // JSON array
    public List<string> Sizes { get; set; } = new(); // parsed from available_sizes
    public string SizeGuideHtml { get; set; } = string.Empty;
    public string Barcode { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
    public bool IsFeatured { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public Category? Category { get; set; }
    public Inventory? Inventory { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}
